### PR TITLE
Fix mask-rcnn region proposal netowrk onnx exporting(issue1741)

### PR DESCRIPTION
**Pull request by shaoboc-lmf**
Related isssue: #1741

1. Change `strides` from int -> tensor
   `strides` in AnchorGenerator's forward needs to be a tensor for onnx exporting
2. Change `torch.clip` -> `torch.min/max`
    torch.clip is used in box_ops.clip_boxes_to_image. Its arguments are always traced as constants. Use torch.min/max to WAR
3. split -> index
  ` torch.split`'s split_size is traced as constant, Use index to WAR

- The `images_sizes` in the `image_list` are intialized as `List(Tuple(int64, int64))`. We should use tensor instead. 
- Above fixes are only for dynamic_axes in H,W dimension. N dimension is not considered yet.

### CODE REVIEW

File: torchvision/models/detection/rpn.py [hunk 1]
```diff
@@ -27,6 +27,56 @@ def _onnx_get_num_anchors_and_pre_nms_top_n(ob, orig_pre_nms_top_n):
 
     return num_anchors, pre_nms_top_n
 
+@torch.jit.unused
+def _onnx_get_grid_image_sizes(feature_maps, image_list_tensors):
+    # type: (List[Tensor], Tensor) -> Tuple(List[Tensor], Tensor)
```
Comment (fmassa) [inline torchvision/models/detection/rpn.py]: I believe there is a typo here:
```python
Typle(List[Tensor], Tensor)
```
should probably read
```python
Tuple[List[Tensor], Tensor]
```

File: torchvision/models/detection/rpn.py [hunk 2]
```diff
@@ -28,6 +28,60 @@ def _onnx_get_num_anchors_and_pre_nms_top_n(ob, orig_pre_nms_top_n):
     return num_anchors, pre_nms_top_n
 
 
+@torch.jit.unused
+def _onnx_get_grid_image_sizes(feature_maps, image_list_tensors):
+    # type: (List[Tensor], Tensor) -> Tuple[List[Tensor], Tensor]
```
Comment (fmassa) [inline torchvision/models/detection/rpn.py]: Although technically incorrect, in order for torchscript to work you'll need to make the return types in the annotation to be consistent with the ones in the other branch, following the error message in https://circleci.com/gh/pytorch/vision/73728?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

So I would instead do something like
```python
type: (List[Tensor], Tensor) -> Tuple[List[List[int]], List[int]]
```
so that `grid_sizes` and `image_sizes` have the "same type" as in the other part of the code

**Comment by lara-hdr** [comment]
@shaoboc-lmf thanks for the PR!
I am curious which input  sizes you tried for testing this code; I tried exporting with (400,200) and then running the model with a size of (300, 500), but it still fails with size errors on an expand node.

**Comment by shaoboc-lmf** [comment]
> @shaoboc-lmf thanks for the PR!
> I am curious which input sizes you tried for testing this code; I tried exporting with (400,200) and then running the model with a size of (300, 500), but it still fails with size errors on an expand node.

I tried on 5 images in coco2017 validation set and just focused on the `backbone + rpn` network only.   
The image_ids of the 5 images I used are: 000000000[139/285/632/724/776].jpg  (The first 5 image from the sequential sampler).  For the `image_list`, which is the input of `backone + rpn`, after transformation, their sizes will result into: 
tensor-----------image_sizes
[1, 3, 800, 1216] [(800, 1201)]
[1, 3, 896, 800]   [(873, 800)]
[1, 3, 800, 1088] [(800, 1060)]
[1, 3, 1088, 800] [(1066, 800)]
[1, 3, 1216, 800] [(1196, 800)]

I didn't see error on expand node. Which line of the torch module code result into the expand node?

[(The test code I used)](https://github.com/shaoboc-lmf/mrcnn-onnx-export/blob/master/test_onnx_backbone_rpn.py)

File: torchvision/models/detection/rpn.py (hunk 1 above)
Comment (shaoboc-lmf) [reply torchvision/models/detection/rpn.py]: Sorry, I may not have time to tidy my tests recently.  I'll do this if I have free cycles to spin.

Comment (fmassa) [reply torchvision/models/detection/rpn.py]: Ok, no worries, thanks for the initial PR!

@lara-hdr do you think someone in your team could help get this PR have a green build, including tests to make sure that RPN now supports arbitrary input shapes?

Comment (lara-hdr) [reply torchvision/models/detection/rpn.py]: Yes, some tests are failing. I’ll do this first thing after the weekend.

**Comment by fmassa** [comment]
This has been fixed in the 0.7.0 release of torchvision https://github.com/pytorch/vision/releases/tag/v0.7.0 thanks for your help!
